src/Model/List/SS_List.php

```diff
@@ -15,83 +16,214 @@
+     *          // SQL: WHERE ("Name" IN ('aziz','bob'))
+     * @example // bob or anyone aged 21 in the list
+     *          $list = $list->filterAny(array('Name'=>'bob, 'Age'=>21));
+     *          // SQL: WHERE ("Name" = 'bob' OR "Age" = '21')
+     * @example // bob or anyone aged 21 or 43 in the list
+     *          $list = $list->filterAny(array('Name'=>'bob, 'Age'=>array(21, 43)));
+     *          // SQL: WHERE ("Name" = 'bob' OR ("Age" IN ('21', '43'))
+     * @example // all bobs, phils or anyone aged 21 or 43 in the list
+     *          $list = $list->filterAny(array('Name'=>array('bob','phil'), 'Age'=>array(21, 43)));
+     *          // SQL: WHERE (("Name" IN ('bob', 'phil')) OR ("Age" IN ('21', '43'))
+     *
+     * @param string|array See {@link filter()}
+     * @return SS_List<T>
+     */
+    public function filterAny(...$args): SS_List;
+
+    /**
+     * Return a new instance of this list that excludes any items with these characteristics
+     *
+     * @example $list = $list->exclude('Name', 'bob'); // exclude bob from list
+     * @example $list = $list->exclude('Name', array('aziz', 'bob'); // exclude aziz and bob from list
+     * @example $list = $list->exclude(array('Name'=>'bob, 'Age'=>21)); // exclude bob that has Age 21
+     * @example $list = $list->exclude(array('Name'=>'bob, 'Age'=>array(21, 43))); // exclude bob with Age 21 or 43
+     * @example $list = $list->exclude(array('Name'=>array('bob','phil'), 'Age'=>array(21, 43)));
+     *          // bob age 21 or 43, phil age 21 or 43 would be excluded
+     *
+     * @return SS_List<T>
+     */
+    public function exclude(...$args): SS_List;
+
+    /**
+     * Return a copy of this list which does not contain any items with any of these params
+     *
+     * @return SS_List<T>
+     */
+    public function excludeAny(...$args): SS_List;
+
+    /**
+     * Return a new instance of this list that excludes any items with these characteristics
+     * Filter this List by a callback function. The function will be passed each record of the List in turn,
+     * and must return true for the record to be included. Returns the filtered list.
+     *
+     * @example $list = $list->filterByCallback(function($item, $list) { return $item->Age == 9; })
+     * @return SS_List<T>
+     */
+    public function filterByCallback(callable $callback): SS_List;
+
+    /**
+     * Return the first item with the given ID
+     *
+     * @param int $id
+     * @return T|null
+     */
+    public function byID(int|string|null $id): mixed;
+    // Note that string ID's and null values should be handled as things like
+    // form submissions and controller params will often be string IDs,
+    // and null is often unintentionally passed in a lot of instances.
+    // If this is changed in the future to only accept int, ensure that all other
+    // filtering methods are updated at the same time for consistency
```

Comment:
This comment placement is weird - it would make sense if a method was implemented here and this was a comment just before some code that normalises the value of that parameter, but that's not the case here.

Ideally the parameter typing should just be clear, and not need a lengthy comment to explain it.
I think I'd prefer just _not_ adding a type here, instead of having this lengthy comment.
If you insist on having a type that requires a lengthy comment, the comment should probably be in the `@param` docblock (which, btw, is currently incorrect in this method, as it says `int` which doesn't match the actual typehint).